src/main/java/com/github/manolo8/darkbot/gui/titlebar/ExtraButton.java

```diff
@@ -31,12 +33,41 @@
     private static final Set<ExtraMenuProvider> EXTRA_DECORATIONS = new LinkedHashSet<>();
     private static final Lazy<Void> clean = new Lazy.NoCache<>();
 
-    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider) {
+    private static final PluginExtraMenuProvider PLUGIN_DECORATIONS = new PluginExtraMenuProvider();
+    private static final Map<Plugin, Set<ExtraMenuProvider>> plugins = new HashMap<>();
+
+    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider, FeatureRegistry featureRegistry) {
         EXTRA_DECORATIONS.clear();
-        provider.forEach(EXTRA_DECORATIONS::add);
+        plugins.clear();
+
+        provider.forEach(extra -> {
+            if (!(extra instanceof DefaultExtraMenuProvider)) {
```

Comment:
The check should be for if this belongs to a plugin, not for if it's the default. In the future the main bot may register more native menu providers (in fact, this PR should be registering a second native menu provider for plugins) so make the check based on that